includes/data-stores/class-wc-order-data-store-cpt.php

```diff
@@ -0,0 +1,557 @@
+	 *
+	 * @param WC_Order|int $order
+	 * @return boolet
+	 */
+	public function get_download_permissions_granted( $order ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		return wc_string_to_bool( get_post_meta( $order_id, '_download_permissions_granted', true ) );
+	}
+
+	/**
+	 * Stores information about whether permissions were generated yet.
+	 *
+	 * @param WC_Order|int $order
+	 * @param bool $set
+	 */
+	public function set_download_permissions_granted( $order, $set ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		update_post_meta( $order_id, '_download_permissions_granted', wc_bool_to_string( $set ) );
+	}
+
+	/**
+	 * Gets information about whether sales were recorded.
+	 *
+	 * @param WC_Order|int $order
+	 * @return boolet
+	 */
+	public function get_recorded_sales( $order ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		return wc_string_to_bool( get_post_meta( $order_id, '_recorded_sales', true ) );
+	}
+
+	/**
+	 * Stores information about whether sales were recorded.
+	 *
+	 * @param WC_Order|int $order
+	 * @param bool $set
+	 */
+	public function set_recorded_sales( $order, $set ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		update_post_meta( $order_id, '_recorded_sales', wc_bool_to_string( $set ) );
+	}
+
+	/**
+	 * Gets information about whether coupon counts were updated.
+	 *
+	 * @param WC_Order|int $order
+	 * @return boolet
+	 */
+	public function get_recorded_coupon_usage_counts( $order ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		return wc_string_to_bool( get_post_meta( $order_id, '_recorded_sales', true ) );
+	}
+
+	/**
+	 * Stores information about whether coupon counts were updated.
+	 *
+	 * @param WC_Order|int $order
+	 * @param bool $set
+	 */
+	public function set_recorded_coupon_usage_counts( $order, $set ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		update_post_meta( $order_id, '_recorded_sales', wc_bool_to_string( $set ) );
+	}
+
+	/**
+	 * Gets information about whether stock was reduced.
+	 *
+	 * @param WC_Order|int $order
+	 * @return boolet
+	 */
+	public function get_stock_reduced( $order ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$order_id = $order->get_id();
+		} else {
+			$order_id = $order;
+		}
+		return wc_string_to_bool( get_post_meta( $order_id, '_order_stock_reduced', true ) );
+	}
+
+	/**
+	 * Stores information about whether stock was reduced.
+	 *
+	 * @param WC_Order|int $order
+	 * @param bool $set
+	 */
+	public function set_stock_reduced( $order, $set ) {
+		if ( is_a( $order, 'WC_Order' ) ) {
```

Comment:
Instead of repeating the order ID check on all these functions, can we make get_order_id on the factory public and use that?